src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -33,6 +38,27 @@ public int Start()
 
             ICommand runCommand = GetRunCommand();
 
+            if (!NoLaunchProfile)
+            {
+                var buildPathContainer = File.Exists(Project) ? Path.GetDirectoryName(Project) : Project;
+                var launchSettingsPath = Path.Combine(buildPathContainer, "Properties", "launchSettings.json");
+                if (File.Exists(launchSettingsPath))
+                {
+                    var launchSettingsFileContents = File.ReadAllText(launchSettingsPath);
+                    if (!LaunchSettingsManager.TryApplyLaunchSettings(launchSettingsFileContents, ref runCommand, out string runAfterLaunch, LaunchProfile))
+                    {
+                        string profileName = string.IsNullOrEmpty(LaunchProfile) ? LocalizableStrings.DefaultLaunchProfileDisplayName : LaunchProfile;
+                        //Error that the launch profile couldn't be applied
+                        Reporter.Error.WriteLine(string.Format(LocalizableStrings.RunCommandExceptionCouldNotApplyLaunchSettings, profileName));
```

Comment:
With the catch-all blocks, lots of programming errors can surface only as "xxx could not be applied", which will be a real pain to diagnose. We should be indicating why we can't apply the settings in situations we expect: invalid json, I/O error, etc., and bugs should be allowed to bubble out of here.